From 8ae4c149de28e7f84c2822c0c2f06c80e731cc62 Mon Sep 17 00:00:00 2001
From: Yan Markman <ymarkman@marvell.com>
Date: Mon, 12 Mar 2018 18:08:02 +0200
Subject: [PATCH 0510/1051] arm64: dts: marvell: add new mcbin-single-shot
 board

Add new Macchiato version board which has single physical
interface on ports 0 -- only Fiber and no Copper 3310 PHY.

The only difference is - no phy property declaration

Change-Id: Ia5876f6371d897c2f29736f623a57f9eb2a69365
Signed-off-by: Yan Markman <ymarkman@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/52205
Reviewed-by: Christine Gharzuzi <chrisg@marvell.com>
Reviewed-by: Victor Gu <xigu@marvell.com>
Tested-by: Victor Gu <xigu@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/59459
Reviewed-by: Stefan Chulski <stefanc@marvell.com>
Tested-by: Stefan Chulski <stefanc@marvell.com>
[Kevin: The original patch got from Marvell sdk10.0_19.06]
Signed-off-by: Kevin Hao <kexin.hao@windriver.com>
---
 arch/arm64/boot/dts/marvell/Makefile          |  1 +
 .../marvell/armada-8040-mcbin-single-shot.dts | 41 +++++++++++++++++++
 2 files changed, 42 insertions(+)
 create mode 100644 arch/arm64/boot/dts/marvell/armada-8040-mcbin-single-shot.dts

diff --git a/arch/arm64/boot/dts/marvell/Makefile b/arch/arm64/boot/dts/marvell/Makefile
index ea9d49f2a911..81d8d8ac0c3a 100644
--- a/arch/arm64/boot/dts/marvell/Makefile
+++ b/arch/arm64/boot/dts/marvell/Makefile
@@ -5,4 +5,5 @@ dtb-$(CONFIG_ARCH_MVEBU) += armada-3720-espressobin.dtb
 dtb-$(CONFIG_ARCH_MVEBU) += armada-7040-db.dtb
 dtb-$(CONFIG_ARCH_MVEBU) += armada-8040-db.dtb
 dtb-$(CONFIG_ARCH_MVEBU) += armada-8040-mcbin.dtb
+dtb-$(CONFIG_ARCH_MVEBU) += armada-8040-mcbin-single-shot.dtb
 dtb-$(CONFIG_ARCH_MVEBU) += armada-8080-db.dtb
diff --git a/arch/arm64/boot/dts/marvell/armada-8040-mcbin-single-shot.dts b/arch/arm64/boot/dts/marvell/armada-8040-mcbin-single-shot.dts
new file mode 100644
index 000000000000..865a03fce65d
--- /dev/null
+++ b/arch/arm64/boot/dts/marvell/armada-8040-mcbin-single-shot.dts
@@ -0,0 +1,41 @@
+// SPDX-License-Identifier: (GPL-2.0+ OR MIT)
+/*
+ * Copyright (C) 2016 Marvell Technology Group Ltd.
+ *
+ * Device Tree for MACCHIATOBin SingleShot Armada 8040 community board platform
+ * Difference from MACCHIATOBin DoubleShot:
+ *   doesn't have two Copper 3310 PHYs connected to port 0 on each CP110
+ */
+
+#include "armada-8040-mcbin.dts"
+
+/ {
+	model = "Marvell 8040 MACHIATOBin SingleShot";
+	compatible = "marvell,armada8040-mcbin-single-shot",
+			"marvell,armada8040-mcbin",
+			"marvell,armada8040", "marvell,armada-ap806-quad",
+			"marvell,armada-ap806";
+
+	cp0 {
+		config-space@f2000000 {
+			cp0_ethernet: ethernet@0 {
+				cp0_eth0: eth0 {
+					/delete-property/phy;
+				};
+			};
+			/delete-node/cp0_xmdio;
+		};
+	};
+
+	cp1 {
+		config-space@f4000000 {
+			cp1_ethernet: ethernet@0 {
+				cp1_eth0: eth0 {
+					/delete-property/phy;
+				};
+			};
+			/delete-node/cp1_xmdio;
+		};
+	};
+};
+
-- 
2.17.1

